### ISSUE TITLE
Implementation questions 

### ISSUE THREAD

**Issue by flav27** [rraileanu/idaac#6]
Hello,

Thank you for sharing the code!
I try to adapt the code for stable-baselines and I have some questions:

1. In the **IDAACnet** **act** method you call **self.base** two times:
    a. _gae, actor_features = self.base(inputs)_
    b. _gae, _ = self.base(inputs, action)_
    Is this necessary, or I could do the following:
    - move the **critic_linear** from **PolicyResNetBase** to **IDAACnet** and modify **act()** like this:
        **actor_features = self.base(inputs)**
        ....
        **gae  = self.critic_linear (torch.cat([actor_features, actions])**
       so that _PolicyResNetBase.forward_ is called only once.

2. Where you create the optimizers you use two lists of parameters for policy and value, is this a performance optimization, or do you want to avoid a possible interference? 

3. Do you think that adding the auxiliary phase of PPG in DAAC will improve results? 

Thank you,
Flaviu

**Comment by rraileanu** [comment]
Hi, apologies for the late response. 

1. I think you are right, you can implement it in the way you mention. I just wanted to keep it similar to the PPO / baseline implementation for consistency. 

3.  Since the policy and value are trained using separate networks / weights, we need different optimizers. Note that Adam is an adaptive optimizer so the parameters will change depending on the weights. Using separate optimizers is cleaner to avoid interference, as you mention. 

4. I haven't tried this, but I believe it shouldn't help in these types of environments, due to the problem explained in the paper. If you train the network to predict the value function, it may learn a representation that overfits to the training levels (e.g. episode lengths) which is not what you want if you care about generalization (and only have a limited number of training  environments). If you apply the auxiliary loss, you are distilling the value into the policy network, which is what we are trying to avoid in the first place.

Let me know if you have any other questions.

**Comment by flav27** [comment]
Thank you for taking the time to respond.

3. I tested DAAC + PPG on my environment and it closes the gap between train and evaluation but in the same environment that has slightly modified episodes, didn't have time/resources to test on procgen.
